### PR TITLE
Fixes for history state due to nesting of services

### LINKED ISSUE #1237: Unskip test "should initialize correctly with preexisting state"
Broken by #1236

### PR DESCRIPTION
**Pull request by dvoytenko**
Closes #1237.

### CODE REVIEW

**Comment by erwinmombay** [comment]
@dvoytenko LGTM

File: test/functional/test-history.js [hunk 1]
```diff
@@ -181,17 +179,21 @@ describe('HistoryBindingNatural', () => {
...
-
-    new HistoryBindingNatural_(window);
-
-    expect(argumentLength).to.equal(2);
-
-    window.history.replaceState = origReplace;
+    const replaceStateSpy = sinon.spy();
+    const windowStub = {
+      history: {
+        replaceState: function() {
```
Comment (jridgewell) [inline test/functional/test-history.js]: You can just use `replaceState: replaceStateSpy`

Comment (dvoytenko) [inline test/functional/test-history.js]: Can't. Because it gets monkey-patched.

Comment (dvoytenko) [inline test/functional/test-history.js]: Actually, no, I'm wrong. This is a good point. Fixed.

**Comment by jridgewell** [comment]
LGTM

**Pull request merged**